GitUI/CommandsDialogs/FormClone.cs

```diff
@@ -67,7 +67,7 @@ protected override void OnRuntimeLoad(EventArgs e)
 
             _NO_TRANSLATE_To.Text = AppSettings.DefaultCloneDestinationPath;
 
-            if (url.IsNotNullOrWhitespace())
+            if (GitModule.PathIsUrl(url) && GitModule.IsValidGitWorkingDir(url))
```

Comment:
Have you seen how the `GitModule.PathIsUrl(url)` is implemented?
I think that changing the operator to OR will break your change. 
The OR operator should stay there but the PathIsUrl should be replaced with sth similar to 
https://github.com/anrichter/gitextensions/blob/83577f61289b8d00858c42f36e10353b834dac69/GitUI/CommandsDialogs/FormClone.cs#L86